Cabal/Distribution/Utils/LogProgress.hs

```diff
@@ -15,21 +17,40 @@ import Distribution.Utils.Progress
 import Distribution.Verbosity
 import Distribution.Simple.Utils
 import Text.PrettyPrint (Doc, (<+>), text, render)
-import Control.Monad (when)
+
+data LogEnv = LogEnv {
+        le_verbosity :: Verbosity,
+        le_context   :: [Doc]
+    }
 
 -- | The 'Progress' monad with specialized logging and
 -- error messages.
-newtype LogProgress a = LogProgress (Progress LogMsg Doc a)
-    deriving (Functor, Applicative, Monad)
+newtype LogProgress a = LogProgress { unLogProgress :: LogEnv -> Progress LogMsg Doc a }
```

Comment:
Cabal doesn't depend on transformers (yet). When parsec becomes default it will but I don't think that GHC's build system has been adjusted to handle this yet.